sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/service/schema/JsonSchemaManagerTest.java

```diff
@@ -0,0 +1,188 @@
+
+  @Inject ObjectMapper objectMapper;
+
+  @Inject DocumentProperties documentProperties;
+
+  @Inject GrpcClients grpcClients;
+
+  @InjectMock StargateRequestInfo requestInfo;
+
+  ArgumentCaptor<QueryOuterClass.Query> queryCaptor;
+
+  @BeforeEach
+  public void init() {
+    queryCaptor = ArgumentCaptor.forClass(QueryOuterClass.Query.class);
+    doAnswer(invocation -> grpcClients.bridgeClient(Optional.empty(), Optional.empty()))
+        .when(requestInfo)
+        .getStargateBridge();
+  }
+
+  Schema.CqlKeyspaceDescribe getValidTableAndKeyspace(String namespace, String collection) {
+    QueryOuterClass.ColumnSpec.Builder partitionColumn =
+        QueryOuterClass.ColumnSpec.newBuilder()
+            .setName(documentProperties.tableProperties().keyColumnName());
+    Set<QueryOuterClass.ColumnSpec> clusteringColumns =
+        documentProperties.tableColumns().pathColumnNames().stream()
+            .map(c -> QueryOuterClass.ColumnSpec.newBuilder().setName(c).build())
+            .collect(Collectors.toSet());
+    Set<QueryOuterClass.ColumnSpec> valueColumns =
+        documentProperties.tableColumns().valueColumnNames().stream()
+            .map(c -> QueryOuterClass.ColumnSpec.newBuilder().setName(c).build())
+            .collect(Collectors.toSet());
+
+    Schema.CqlTable.Builder table =
+        Schema.CqlTable.newBuilder()
+            .setName(collection)
+            .putOptions("comment", "{\"schema\": {}}")
+            .addPartitionKeyColumns(partitionColumn)
+            .addAllClusteringKeyColumns(clusteringColumns)
+            .addAllColumns(valueColumns);
+    return Schema.CqlKeyspaceDescribe.newBuilder()
+        .setCqlKeyspace(Schema.CqlKeyspace.newBuilder().setName(namespace))
+        .addTables(table)
+        .build();
+  }
+
+  @Nested
+  class AttachJsonSchema {
+    @Test
+    public void happyPath() throws JsonProcessingException {
+      String namespace = RandomStringUtils.randomAlphanumeric(16);
+      String collection = RandomStringUtils.randomAlphanumeric(16);
+
+      QueryOuterClass.Response response = QueryOuterClass.Response.newBuilder().build();
+
+      doAnswer(
+              invocationOnMock -> {
+                StreamObserver<QueryOuterClass.Response> observer = invocationOnMock.getArgument(1);
+                observer.onNext(response);
+                observer.onCompleted();
+                return null;
+              })
+          .when(bridgeService)
+          .executeQuery(any(), any());
+
+      JsonNode schema =
+          objectMapper.readTree("{\"$schema\": \"https://json-schema.org/draft/2019-09/schema\"}");
+      jsonSchemaManager
+          .attachJsonSchema(namespace, collection, schema)
+          .subscribe()
+          .withSubscriber(UniAssertSubscriber.create())
+          .awaitItem();
+
+      verify(bridgeService).executeQuery(any(), any());
+      verifyNoMoreInteractions(bridgeService);
+    }
+  }
+
+  @Nested
+  class GetJsonSchema {
```

Comment:
missing cases:

* no schema in table
* comment not a valid schema (imo should behave as no schema on table, or?)